### ISSUE TITLE
allow for additional plotting options

### ISSUE THREAD

**Issue by wds15** [stan-dev/bayesplot#45]
It would be great to let users customize more the plots from bayesplot to make them ready for presentations or other non-screen usage.

A functionality like rstan_gg_options would be great to have which allows for specification of options which are beyond ggplot theme options and go directly to the used geoms.

**Comment by jgabry** [comment]
I agree that would be good to do. I might not get to it before the initial
CRAN release but definitely on the short term to-do list.

**Comment by jgabry** [comment]
@wds15 Which particular additional options do you think would be most useful?

**Comment by wds15** [comment]
Argh... these notifications... sorry for the delay.

The main goal is to be able to adjust for using the plots in different contexts. Namely:

- data exploration on screen - most often small sizes
- presentation - things need to be large and bold
- publication - few colours, moderate size

So I guess we need point/line/font sizes for this to happen. Optimally the three use-cases I outlined are easily set by the user via an example in the help or similar.

**Comment by bob-carpenter** [comment]
The moderate size thing is the trickiest.  Ideally we'd
just keep PDFs, but R generates PDFs proportional in size
to the number of data points for things like scatter plots
or line plots.  

I think you'll also find that each publication has different
standards, so that may not be a reasonable thing to try to
anticipate in advance.

I don't agree about presentations needing to be bold if
you mean the font weight.  But everything needs to be readable
by the audience.

And think about us old farts.  Don't use 6pt fonts, even for
exploration!

- Bob

**Comment by jsta** [comment]
Maybe I just can't find it but there doesn't seem to be a way to change the legend text on `ppc_intervals` to anything but `y` and `y_{rep}`.

**Comment by mcol** [comment]
@jsta I've opened issue #157 for this.

**Comment by avehtari** [comment]
>  but definitely on the short term to-do list.

Ping @jgabry

I'm using mcmc_scatter as it plots also divergences, but changing alpha doesn't work for the default shape=19, and I really would like to be able to change the shape. It is possible to change the shape of the divergence points, but not the main points.

**Comment by jgabry** [comment]
> I'm using mcmc_scatter as it plots also divergences, but changing alpha doesn't work for the default shape=19, and I really would like to be able to change the shape. It is possible to change the shape of the divergence points, but not the main points.

@behramulukir maybe this is something you could add if you're interested?

**Comment by behramulukir** [comment]
Yes, I can take care of it. I think adding a set of optional parameters to `scatter_style_np` to control the visuals of the main points would be enough, right?

**Comment by jgabry** [comment]
Thanks @behramulukir. I think in this case you'd want to add a `shape` argument to `mcmc_scatter` itself. The `scatter_style_np` function is intended to just control the appearance of the NUTS sample parameters that are plotted (e.g. the divergences).

**Comment by behramulukir** [comment]
Alright, makes sense. I'll work on this
